src/slug.ts

```diff
@@ -0,0 +1,19 @@
+export function slugref(envVar: string): string {
+  return envVar
+    .toLowerCase()
+    .replace(RegExp('refs/(heads|tags)/'), '')
+    .replace(new RegExp('[_/@]', 'g'), '-')
+    .substring(0, 63)
+}
+
+export function slugurl(envVar: string): string {
+  return slugref(envVar).replace(new RegExp('[.]', 'g'), '-')
+}
+
+export function slugurlref(envVar: string): string {
+  return slugurl(slugref(envVar))
```

Comment:
since slugurl call slugref, no need to call slugref, you can call slugurl directly